discovery-infra/tests/test_logs.py

```diff
@@ -0,0 +1,91 @@
+        self.wait_for_cluster_to_install(cluster_id=cluster_id, api_client=api_client)
+
+    def test_collect_logs_on_success(self, api_client, node_controller, cluster):
+        # Define new cluster and prepare it for installation
+        cluster_id = cluster().id
+        self.setup_hosts(cluster_id, api_client, node_controller)
+        self.prepare_for_installation(cluster_id, api_client, node_controller)
+        
+        #download logs into a file. At this point logs are not uploaded
+        path = "/tmp/test_on-success_logs.tar"
+        self.verify_no_logs_uploaded(cluster_id, api_client, path)
+        
+        #install the cluster
+        ts_start_at = time.time()
+        self.install_cluster_and_wait(cluster_id, api_client)
+
+        #download logs into a file. At this point logs exist
+        expected_min_log_num = len(node_controller.list_nodes()) + 1
+        self.verify_logs_uploaded(cluster_id, api_client, path, ts_start_at, expected_min_log_num)
+
+    
+    def test_collect_logs_on_failure(self, api_client, node_controller, cluster):
+        '''cacnel insllation after at least one host is booted and check that logs are uploaded'''
+        cluster_id = cluster().id
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+        node_controller.start_all_nodes()
+        self.wait_until_hosts_are_discovered(cluster_id=cluster_id, api_client=api_client)
+        self.set_host_roles(cluster_id=cluster_id, api_client=api_client)
+        self.set_network_params(
+            cluster_id=cluster_id,
+            api_client=api_client,
+            controller=node_controller
+        )
+
+        ts_start_at = time.time()
+        self.start_cluster_install(cluster_id=cluster_id, api_client=api_client)
+        # Cancel cluster install once at least one host booted
+        self.wait_for_one_host_to_boot_during_install(cluster_id=cluster_id, api_client=api_client)
+        self.cancel_cluster_install(cluster_id=cluster_id, api_client=api_client)
+        assert self.is_cluster_in_cancelled_status(
+            cluster_id=cluster_id,
+            api_client=api_client
+        )
+         #download logs into a file. At this point logs exist
+        path = "/tmp/test_on_cancel_logs.tar"
+        expected_min_log_num = 1 #at least the booted master has logs
+        self.verify_logs_uploaded(cluster_id, api_client, path, ts_start_at, expected_min_log_num)
+
+    def verify_no_logs_uploaded(self, cluster_id, api_client, path):
+        with pytest.raises(ApiException) as ex:
+            api_client.download_cluster_logs(cluster_id, path)
+
+    def verify_logs_uploaded(self, cluster_id, api_client, path, ts_start_at, expected_min_log_num):
+        api_client.download_cluster_logs(cluster_id, path)
+        tar = tarfile.open(path)
+        loglist = []
+        for tarinfo in tar:
+            loglist.append(tarinfo.name)
+            assert tarinfo.mtime > ts_start_at
```

Comment:
You're just verifying when the tar items has been created. @tsorya isn't that going to be always up-to-date to when we download the logs? That's when the tar is being created.
So IMO this condition doesn't actually verify anything.
Maybe we can check last "updated_at" value of the last log to know when it was last updated.

Reply:
can we expect number of files?

Reply:
@YuviGold i think it will be too much to validate log line. we can though validate log_uploaded_at flag per host and controller